### ISSUE TITLE
Database split

### ISSUE THREAD

**Issue by damiansm** [exomiser/Exomiser#161]
More detail on db split:

Splitting the database. We’re going to try and split it so that the prioritiser package can run independently of the rest of the application. The other advantage of this is we can then provide both hg37 and hg38 data in the same release and independently release changes for the other data. To do this we’ll: a. Create at least two h2 databases – variant and phenotype plus maybe a ‘common’ one containing shared data, like diseases, models, gene symbols.
b. Probably the next release we’ll look into changing the database back-ends so that they use a suitable engine for the type of data stored – document for the variant data, graph for the phenotype data and probably relational for the common.
c. Update the core and spring-boot-autoconfigure libraries to handle the split.
d. The split and library updates will have to happen together, but it will make the database backend changes easier as a second step.
e. We’re going to have to think of some sensible naming conventions for these – possibly {name}-{compatible-version}-{build-date} e.g. variant-8.0.0-20161104
f. In order not to have inconsistent gene ids between what’s in Jannovar and the actual database we’re going to need some way of freezing/synchronising the right bits of this data.

**Comment by julesjacobsen** [comment]
Related to #187

**Comment by damiansm** [comment]
Notes from new phenotype-build experiments:

- Initial attempt to use Monarch data-sources documented at test_monarch_data_extraction.pl but largely abandoned. Still follow up with Kent to see if can do in future.
- For now wrote new parsers and worked out new OWLSim commands:
- install owltools from git clone https://github.com/owlcollab/owltools.git and add owltools and owltools-runner to path
```
git clone https://github.com/monarch-initiative/monarch-owlsim-data
git clone https://github.com/obophenotype/human-phenotype-ontology/
git clone https://github.com/obophenotype/mammalian-phenotype-ontology/
cd mammalian-phenotype-ontology/src/ontology
make mp.owl (OR CAN JUST wget from MGI: wget http://www.informatics.jax.org/downloads/reports/mp.owl)
git clone https://github.com/obophenotype/upheno
wget http://compbio.charite.de/jenkins/job/zp-owl/lastSuccessfulBuild/artifact/zp.owl

RUN produce_new_disease_data_files.pl to replace the data files for Homo_sapiens with just RD annotations from HPO site as Monarch ones include common disease and merge various OMIM and Orphanet entries together

bsub -q bio -o make_mouse.txt -e make_mouse.err 'owltools --catalog-xml upheno/catalog-v001.xml mammalian-phenotype-ontology/scratch/mp-importer.owl mammalian-phenotype-ontology/src/ontology/mp.owl human-phenotype-ontology/hp.owl zp.owl monarch-owlsim-data/data/Mus_musculus/Mm_gene_phenotype.txt monarch-owlsim-data/data/Homo_sapiens/Hs_disease_phenotype.txt monarch-owlsim-data/data/Danio_rerio/Dr_gene_phenotype.txt --merge-imports-closure --load-instances monarch-owlsim-data/data/Mus_musculus/Mm_gene_phenotype.txt --load-labels monarch-owlsim-data/data/Mus_musculus/Mm_gene_labels.txt --merge-support-ontologies -o Mus_musculus-all.owl'

bsub -q bio -o make_human.txt -e make_human.err 'owltools --catalog-xml upheno/catalog-v001.xml human-phenotype-ontology/scratch/hp-importer.owl mammalian-phenotype-ontology/src/ontology/mp.owl human-phenotype-ontology/hp.owl zp.owl monarch-owlsim-data/data/Mus_musculus/Mm_gene_phenotype.txt monarch-owlsim-data/data/Homo_sapiens/Hs_disease_phenotype.txt monarch-owlsim-data/data/Danio_rerio/Dr_gene_phenotype.txt --merge-imports-closure --load-instances monarch-owlsim-data/data/Homo_sapiens/Hs_disease_phenotype.txt --load-labels monarch-owlsim-data/data/Homo_sapiens/Hs_disease_labels.txt --merge-support-ontologies -o Homo_sapiens-all.owl'

bsub -q bio -o make_fish.txt -e make_fish.err 'owltools --catalog-xml upheno/catalog-v001.xml upheno/vertebrate.owl mammalian-phenotype-ontology/src/ontology/mp.owl human-phenotype-ontology/hp.owl zp.owl monarch-owlsim-data/data/Mus_musculus/Mm_gene_phenotype.txt monarch-owlsim-data/data/Homo_sapiens/Hs_disease_phenotype.txt monarch-owlsim-data/data/Danio_rerio/Dr_gene_phenotype.txt --load-instances monarch-owlsim-data/data/Danio_rerio/Dr_gene_phenotype.txt --load-labels monarch-owlsim-data/data/Danio_rerio/Dr_gene_labels.txt --load-instances monarch-owlsim-data/data/Homo_sapiens/Hs_disease_phenotype.txt --load-labels monarch-owlsim-data/data/Homo_sapiens/Hs_disease_labels.txt --merge-support-ontologies --merge-imports-closure --remove-disjoints --remove-equivalent-to-nothing-axioms --run-reasoner -r elk --assert-implied --make-super-slim HP,ZP -o hp-zp-all.owl'

bsub -q bio -o merge.txt -e merge.err 'owltools Homo_sapiens-all.owl --merge-import-closure --remove-disjoints --remove-equivalent-to-nothing-axioms -o Homo_sapiens-all-merged.owl'
bsub -q bio -o merge.txt -e merge.err 'owltools Mus_musculus-all.owl --merge-import-closure --remove-disjoints --remove-equivalent-to-nothing-axioms -o Mus_musculus-all-merged.owl'
bsub -q bio -o merge.txt -e merge.err 'owltools hp-zp-all.owl --merge-import-closure --remove-disjoints --remove-equivalent-to-nothing-axioms -o hp-zp-all-merged.owl'

bsub -q bio -R 'rusage[mem=100000]' -o hp_hp.txt -e hp_hp.err 'OWLTOOLS_MEMORY=100G owltools Homo_sapiens-all-merged.owl --sim-save-phenodigm-class-scores -m 2.5 -x HP,HP -a hp-hp-phenodigm-cache.txt'

(not needed) bsub -q bio -R 'rusage[mem=100000]' -o mp_mp.txt -e mp_mp.err 'OWLTOOLS_MEMORY=100G owltools Mus_musculus-all-merged.owl --sim-save-phenodigm-class-scores -m 2.5 -x MP,MP -a mp-mp-phenodigm-cache.txt'

bsub -q bio -R 'rusage[mem=100000]' -o hp_mp.txt -e hp_mp.err 'OWLTOOLS_MEMORY=100G owltools Mus_musculus-all-merged.owl Homo_sapiens-all-merged.owl upheno/hp-mp/mp_hp-align-equiv.owl --merge-support-ontologies --sim-save-phenodigm-class-scores -m 2.5 -x HP,MP -a hp-mp-phenodigm-cache.txt'

bsub -q bio -R 'rusage[mem=100000]' -o hp_zp.txt -e hp_zp.err 'OWLTOOLS_MEMORY=100G owltools hp-zp-all-merged.owl --sim-save-phenodigm-class-scores -m 2.5 -x HP,ZP -a hp-zp-phenodigm-cache.txt'
```
- Test new files with compare_phenodigm_replacement_code.pl
